src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/Wedge.cpp

```diff
@@ -0,0 +1,484 @@
+  // be false even when cos_theta_y >= root_2_over_2 causing the subsequent ifs
+  // to be evaluated even though though they will never be true.
+  if (cos_theta_x >= root_2_over_2) {
+    // Point is in one of +z,+y,-y wedge. Check cos_theta_y now.
+    if (cos_theta_y >= root_2_over_2) {
+      // Point is in +z wedge
+      if (direction_ != Direction<3>::upper_zeta()) {
+        return std::nullopt;
+      }
+    } else if (target_coords[1] > 0.0) {
+      // Point is in +y wedge
+      if (direction_ != Direction<3>::upper_eta()) {
+        return std::nullopt;
+      }
+    } else if (target_coords[1] < 0.0) {
+      // Point is in -y wedge
+      if (direction_ != Direction<3>::lower_eta()) {
+        return std::nullopt;
+      }
+    } else {
+      ERROR("Inconsistent angle +z,+y,-y");
+    }
+  } else if (cos_theta_x <= -root_2_over_2) {
+    // Point is in one of -z,+y,-y wedge. Check cos_theta_y now.
+    if (cos_theta_y <= -root_2_over_2) {
+      // Point is in -z wedge
+      if (direction_ != Direction<3>::lower_zeta()) {
+        return std::nullopt;
+      }
+    } else if (target_coords[1] > 0.0) {
+      // Point is in +y wedge
+      if (direction_ != Direction<3>::upper_eta()) {
+        return std::nullopt;
+      }
+    } else if (target_coords[1] < 0.0) {
+      // Point is in -y wedge
+      if (direction_ != Direction<3>::lower_eta()) {
+        return std::nullopt;
+      }
+    } else {
+      ERROR("Inconsistent angle -z,+y,-y");
+    }
+  } else if (cos_theta_x < root_2_over_2 and cos_theta_x > -root_2_over_2) {
+    // Point is in one of +y,-y,+x,-x. Check phi_z now.
+    const double phi_z = atan2(target_coords[1], target_coords[0]);
+    const double three_pi_over_four = 3.0 * M_PI_4;
+    if (phi_z >= -M_PI_4 and phi_z <= M_PI_4) {
+      // Point is in +x wedge
+      if (direction_ != Direction<3>::upper_xi()) {
+        return std::nullopt;
+      }
+    } else if (phi_z >= three_pi_over_four or phi_z <= -three_pi_over_four) {
+      // Point is in -x wedge
+      if (direction_ != Direction<3>::lower_xi()) {
+        return std::nullopt;
+      }
+    } else if (target_coords[1] > 0.0) {
+      // Point is in +y wedge
+      if (direction_ != Direction<3>::upper_eta()) {
+        return std::nullopt;
+      }
+    } else if (target_coords[1] < 0.0) {
+      // Point is in -y wedge
+      if (direction_ != Direction<3>::lower_eta()) {
+        return std::nullopt;
+      }
+    } else {
+      ERROR("Inconsistent angle +y,-y,+x,-x");
+    }
+  } else {
+    ERROR("Inconsistent angles.");
+  }
+
+  // We are in the proper wedge direction. It is safe to assume the coordinates
+  // can be rotated from their original orientation to now "aligned" with +z.
+  // Therefore we can continue with computing the original radius.
+  const std::array<double, 3> rotated_coords =
+      discrete_rotation(orientation_map_.inverse_map(), target_coords);
```

Comment:
You could rotate first, and then you only have to check if you're in the +z wedge.